staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -153,24 +153,22 @@ func (rl *respLogger) Addf(format string, data ...interface{}) {
 	rl.addedInfo += "\n" + fmt.Sprintf(format, data...)
 }
 
-// Log is intended to be called once at the end of your request handler, via defer
-func (rl *respLogger) Log() {
+func (rl *respLogger) LogArgs() []interface{} {
 	latency := time.Since(rl.startTime)
-	if klog.V(3).Enabled() {
-		if !rl.hijacked {
-			klog.InfoDepth(1, fmt.Sprintf("verb=%q URI=%q latency=%v resp=%v UserAgent=%q srcIP=%q: %v%v",
-				rl.req.Method, rl.req.RequestURI,
-				latency, rl.status,
-				rl.req.UserAgent(), rl.req.RemoteAddr,
-				rl.statusStack, rl.addedInfo,
-			))
-		} else {
-			klog.InfoDepth(1, fmt.Sprintf("verb=%q URI=%q latency=%v UserAgent=%q srcIP=%q: hijacked",
-				rl.req.Method, rl.req.RequestURI,
-				latency, rl.req.UserAgent(), rl.req.RemoteAddr,
-			))
-		}
+	args := []interface{}{
+		"verb", rl.req.Method,
+		"URI", rl.req.RequestURI,
+		"latency", latency,
+		"resp", rl.status,
+		"userAgent", rl.req.UserAgent(),
+		"srcIP", rl.req.RemoteAddr,
+	}
+	if rl.hijacked {
+		args = append(args, "hijacked", true)
+	}else{
+		args = append(args, rl.statusStack, rl.addedInfo)
```

Comment:
```suggestion
		args = append(args, "resp", rl.status,  rl.statusStack, rl.addedInfo)
		if len(rl.statusStack) > 0 {
		    args = append(args, "statusStack", rl.statusStack)
		}
		if len(rl.addedInfo) > 0 {
		    args = append(args, "addedInfo", rl.addedInfo)
		}
```

Reply:
This log message generates lots of volume, so we should have on keys if possible. 

"statusStack" and "addedInfo" are not frequently used (At least I haven't seen them used in gce performance tests) so let's skip them if they are empty